### DISCUSSION TITLE
Add "svg" option to asset types defined in the "Restrict allowed file types" setting of asset field

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#13522] [Ideas] [answered]
There are sometimes cases when we would like allow only for uploading svg files. This could be because svg can ba handled differently that raster graphics in twig templates.

In case some asset field already exists with "image" option selected in "Restrict allowed file types", new "svg" setting could be automatically selected as well, to avoid changing behavior of the field (if svg option suddenly appeared and was not selected, users would loose ability to upload svg files which were earlier sonsidered part of "image" option).

**Comment by brandonkelly** [comment, marked as answer]
You can add your own custom “SVG” file kind using the [`extraFileKinds`](https://craftcms.com/docs/4.x/config/general.html#extrafilekinds) config setting:

```php
->extraFileKinds([
    'svg' => [
        'extensions' => ['svg'],
    ],
])
```

**Comment by piotrpog** [comment]
@brandonkelly 
Thanks!
